schema.graphql

```diff
@@ -28,13 +30,21 @@ type Order @entity {
   "Trade's OrderUid to hex string"
   id: ID!
   "Trade's owner"
-  owner: User!
-  "block's timestamp"
-  timestamp: BigInt!
+  tradesOwner: User
+  "Presign User"
+  presignUser: User
+  "block's timestamp on trade event"
+  tradesTimestamp: BigInt
+  "block's timestamp on invalidate event"
+  invalidateTimestamp: BigInt
+  "block's timestamp on presign event"
+  presignTimestamp: BigInt
   "Array of trades on the order"
-  trades: [Trade!]! @derivedFrom(field: "order")
+  trades: [Trade!] @derivedFrom(field: "order")
   "Boolean value true by default unless is invalidated by the event"
   isValid: Boolean!
+  "Boolean value false by default unless is presign by the event"
+  isPresigned: Boolean!
```

Comment:
I think it could make more readable queries, but I can remove it.

Reply:
Plmk if you want me to remove it?

Reply:
Again, I'm not against it, just habit of removing duplicate data whenever possible.

If this makes querying easier then we can have :)

Reply:
Removed for now. If we find it is necessary we can add it again later.

Reply:
> As far as I understand we can easily derive this if from presignTimestamp.

It depends. It could be implemented like that, but that's not in this PR at least. 
It's important to note that the presign is a flag the trader sets **or unsets** any time. 


So, we would need `isPresigned` and this could be `false` (default), `true` when the user sets it to true. But the user could change it back to `false` any time.

Alternatively, and if we want to not include this flag and use the timestamp for this, we would need to remove the timestamp when the user is seting the flag to `false` and add the timestamp when this is `true`.

1st option is closer to the contract and has slightly more info, so in this sense is nicer. 2nd option is cleaner and probably good enough.